scipy/misc/common.py

```diff
@@ -116,7 +116,8 @@ def derivative(func, x0, dx=1.0, n=1, args=(), order=3):
     val = 0.0
     ho = order >> 1
     for k in range(order):
-        val += weights[k]*func(x0+(k-ho)*dx,*args)
+        if weights[k] != 0:  # skip unnecessary func() calls
```

Comment:
In preference to this I'd just remove all the zeros from the n==1 weight arrays, also amending line 102 to remove the zero entry returned by central_diff_weights. That will remove the if statement.

It should be straightforward to vectorise the 'for' loop:

1) calculate the delta_x' on the positive side, concatenate with itself (multiplied by -1). This will give `(k - ho) * dx`, but without the zero
2) Add the delta_x' to x0 to get the full list of locations
3) Use a list comprehension to get val